lib/api/CSerializableToJson.cc

```diff
@@ -129,18 +130,42 @@ CSerializableFromCompressedChunkedJson::rawJsonStream(const std::string& compres
                                                       TIStreamPtr inputStream,
                                                       std::iostream& buffer) {
     if (inputStream != nullptr) {
-        core::CRapidJsonUnbufferedIStreamWrapper isw{*inputStream};
+        //        core::CBoostJsonUnbufferedIStreamWrapper isw{*inputStream};
+        //        LOG_DEBUG(<< "inputStream: " << inputStream->rdbuf());
         try {
-            rapidjson::Document doc;
+            json::value doc;
+            json::error_code ec;
+            json::stream_parser p;
+            std::string line;
             bool done{false};
-            do {
-                doc.ParseStream<rapidjson::kParseStopWhenDoneFlag>(isw);
-                assertNoParseError(doc);
-                auto chunk = ifExists(compressedDocTag, getAsObjectFrom, doc);
-                buffer.write(ifExists(payloadTag, getAsStringFrom, chunk),
-                             ifExists(payloadTag, getStringLengthFrom, chunk));
-                done = chunk.HasMember(JSON_EOS_TAG);
-            } while (done == false);
+            while (inputStream->eof() == false && done == false) {
+                if (inputStream->peek() == '\0') {
+                    inputStream->get();
+                    continue;
+                }
+                std::getline(*inputStream, line);
+                LOG_DEBUG(<< "line: " << line);
+                std::size_t length{line.length()};
+                std::size_t written{0};
+                p.reset();
+                while (written < length) {
+                    written += p.write_some(line, ec);
+                    assertNoParseError(ec);
+                }
+                doc = p.release();
+                assertIsJsonObject(doc);
+                LOG_DEBUG(<< "doc: " << doc);
+                try {
+                    auto chunk = ifExists(compressedDocTag, getAsObjectFrom,
+                                          doc.as_object());
+                    buffer.write(ifExists(payloadTag, getAsStringFrom, chunk),
+                                 ifExists(payloadTag, getStringLengthFrom, chunk));
+                    done = chunk.contains(JSON_EOS_TAG);
+                } catch (const std::runtime_error& e) {
+                    LOG_WARN(<< "Caught exception: " << e.what());
```

Comment:
This isn't great. It's catching and silently ignoring errors like `std::bad_alloc`. But I know it's done like this elsewhere in the code, so no need to fix in this PR. As a future project it would be good to review exception handling in the code. The style guide says not to use exceptions but everyone ignores that now. So it ought to be done with a bit more thought than just assuming all exceptions mean the same thing when we catch them.